src/main/java/bio/terra/service/dataset/flight/ingest/IngestPopulateFileStateFromFlightMapStep.java

```diff
@@ -51,7 +56,16 @@ public StepResult doSkippableStep(FlightContext context) {
     List<String> errors = new ArrayList<>();
     try (var bulkFileLoadModels = getModelsStream(ingestRequest, fileColumns, errors)) {
 
-      loadService.populateFiles(loadId, bulkFileLoadModels, batchSize);
+      List<FileModel> existingFiles = new ArrayList<>();
+
+      if (ingestRequest.isResolveExistingFiles()) {
+        var toIngest =
+            bulkFileLoadModels.flatMap(fileModel -> fileNotIngested(fileModel, existingFiles));
```

Comment:
To me, this seems like a mis-use of `Stream.flatMap()`; it's being used to avoid a `null` check. if `fileNotIngested() returned `null` you could write
```suggestion
        var toIngest =
            bulkFileLoadModels.map(fileModel -> fileNotIngested(fileModel, existingFiles))
                .filter(Objects::nonNull);
```